shopinvader_customer_validate/migrations/14.0.1.2.3/pre-migration.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from openupgradelib import openupgrade
+
+# pylint: disable=odoo-addons-relative-import
+from odoo.addons.shopinvader_customer_validate.hooks import (
+    NEW_MODULE,
+    OLD_MODULE,
+    RECORD_XMLIDS,
+    update_record_xmlids,
+)
+
+
+def check_installed(cr):
+    query = """
+        SELECT 1
+        FROM ir_model_data
+        WHERE module = %(old_module)s
+        AND name in %(names)s;
+    """
+    cr.execute(query, {"old_module": OLD_MODULE, "names": tuple(RECORD_XMLIDS)})
+    return cr.fetchall()
+
+
+def delete_new_records(cr):
+    xmlids = [f"{NEW_MODULE}.{xmlid}" for xmlid in RECORD_XMLIDS]
+    openupgrade.delete_records_safely_by_xml_id(xmlids)
+
+
+def migrate(cr, version):
+    # The install hook was missing, some mail templates has been created.
+    # If shopinvader_notification_default was installed, those records were already there
+    # and should have been moved in this module instead.
+    # Don't do anything if there's no remaining records for
+    # shopinvader_notification_default
+    if check_installed(cr):
+        # Delete the new mail.templates / ir.ui.views
+        delete_new_records(cr)
```

Comment:
why do you delete them instead of renaming them?

Reply:
Well, we have records coming from `shopinvader_notification_default`, now we have new records coming from `shopinvader_customer_validate`, all with the same `ir_model_data.name`.
Therefore, it might (should) crash if i try to give an old record the same xmlid as another one (or at least I expect a unicity constraint in the db).
So before renaming `shopinvader_notification_default` records, I have to remove the `shopinvader_customer_validate` ones.